src/ios/LocalyticsPlugin.m

```diff
@@ -423,191 +427,286 @@ - (void)updateInboxCampaignCache {
+        } else {
+            NSLog(@"Localytics Cordova wrapper received bad input in call to tagEvent; Event name is invalid.");
         }
     }
 }
 
 - (void)tagPurchased:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"tagPurchased" withCommand:command];
     if ([command.arguments count] == 5) {
         NSString *itemName = [command argumentAtIndex:0];
         NSString *itemId = [command argumentAtIndex:1];
         NSString *itemType = [command argumentAtIndex:2];
         NSNumber *itemPrice = [command argumentAtIndex:3];
         NSDictionary *attributes = [command argumentAtIndex:4];
         [Localytics tagPurchased:itemName itemId:itemId itemType:itemType itemPrice:itemPrice attributes:attributes];
+    } else {
+        NSLog(@"Localytics Cordova wrapper received bad input in call to tagPurchased; 5 arguments required.");
     }
 }
 
 - (void)tagAddedToCart:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"tagAddedToCart" withCommand:command];
     if ([command.arguments count] == 5) {
         NSString *itemName = [command argumentAtIndex:0];
         NSString *itemId = [command argumentAtIndex:1];
         NSString *itemType = [command argumentAtIndex:2];
         NSNumber *itemPrice = [command argumentAtIndex:3];
         NSDictionary *attributes = [command argumentAtIndex:4];
         [Localytics tagAddedToCart:itemName itemId:itemId itemType:itemType itemPrice:itemPrice attributes:attributes];
+    } else {
+        NSLog(@"Localytics Cordova wrapper received bad input in call to tagAddedToCart; 5 arguments required.");
     }
 }
 
 - (void)tagStartedCheckout:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"tagStartedCheckout" withCommand:command];
     if ([command.arguments count] == 3) {
         NSNumber *totalPrice = [command argumentAtIndex:0];
         NSNumber *itemCount = [command argumentAtIndex:1];
         NSDictionary *attributes = [command argumentAtIndex:2];
         [Localytics tagStartedCheckout:totalPrice itemCount:itemCount attributes:attributes];
+    } else {
+        NSLog(@"Localytics Cordova wrapper received bad input in call to tagStartedCheckout; 3 arguments required.");
     }
 }
 
 - (void)tagCompletedCheckout:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"tagCompletedCheckout" withCommand:command];
     if ([command.arguments count] == 3) {
         NSNumber *totalPrice = [command argumentAtIndex:0];
         NSNumber *itemCount = [command argumentAtIndex:1];
         NSDictionary *attributes = [command argumentAtIndex:2];
         [Localytics tagCompletedCheckout:totalPrice itemCount:itemCount attributes:attributes];
+    } else {
+        NSLog(@"Localytics Cordova wrapper received bad input in call to tagCompletedCheckout; 3 arguments required.");
     }
 }
 
 - (void)tagContentViewed:(CDVInvokedUrlCommand *)command {
+    [self logInput:@"tagContentViewed" withCommand:command];
```

Comment:
Should a similar log be added to the Android bindings?

Reply:
It exists - look right after the beginning of `execute`